src/plugins/workspace/public/components/workspace_form/fields/workspace_description_field.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiCompressedFormRow, EuiCompressedTextArea, EuiTextColor } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import React, { useCallback } from 'react';
+
+import { MAX_WORKSPACE_DESCRIPTION_LENGTH } from '../../../../common/constants';
+
+export interface WorkspaceDescriptionFieldProps {
+  value?: string;
+  onChange: (newValue: string) => void;
+  error?: string;
+  readOnly?: boolean;
+}
+
+export const WorkspaceDescriptionField = ({
+  value,
+  error,
+  readOnly,
+  onChange,
+}: WorkspaceDescriptionFieldProps) => {
+  const handleChange = useCallback(
+    (e) => {
+      onChange(e.currentTarget.value);
+    },
+    [onChange]
+  );
+  const leftCharacters = MAX_WORKSPACE_DESCRIPTION_LENGTH - (value?.length ?? 0);
+  const charactersOverflow = leftCharacters < 0;
```

Comment:
Nit: just wondering if we can use a reusable component for both `workspace_description_field` and `workspace_name_field`.

Reply:
The name field and description field are using different component. The name field is using `EuiCompressedFieldText` and the description is using`EuiCompressedTextArea`. It's a little bit hard to separate a resuable component. Will be separate one if it been used in other places in the future.